### PR TITLE
Add a random number generation tutorial

### LINKED ISSUE #2436: Randomness tutorial
We should create a dedicated docs page for randomness.

See discussion in https://github.com/godotengine/godot/issues/28689, especially the comments [here](https://github.com/godotengine/godot/issues/28689#issuecomment-489464442) and [here](https://github.com/godotengine/godot/issues/28689#issuecomment-489468200).

### PR DESCRIPTION
**Pull request by Calinou**
This closes #2436.

### CODE REVIEW

File: tutorials/math/random_number_generation.rst [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+========================
+
+Many games rely on randomness to implement core game mechanics. This tutorial
+guides you through common types of randomness and how to implement them in
+Godot.
+
+.. note::
+
+    Computers cannot generate "true" random numbers. Instead, they rely on
+    *psuedorandom number generators* (PRNGs).
```
Comment (NathanLovato) [inline tutorials/math/random_number_generation.rst:13]: How about linking to an article that explains that?

File: tutorials/math/random_number_generation.rst [hunk 2]
```diff
@@ -0,0 +1,295 @@
...
+        # To use a string as a seed, you can hash it to a number.
+        seed("Hello world".hash())
+
+When using the RandomNumberGenerator class, you should call ``randomize()``
+on the instance since it has its own seed::
+
+    var rng = RandomNumberGenerator.new()
+    rng.randomize()
+
+Get a random number
```
Comment (NathanLovato) [inline tutorials/math/random_number_generation.rst:61]: In part, this section paraphrases or duplicates the class reference's content. I'd invite people to go read the page. While I understand it's nice to have all the explanations on one page, in a reference manual, I'd try to push people to read the code reference.

It's not that I find this section bad; instead, I feel that's the perfect amount of info and detail we should have in the class ref itself.

**Comment by NathanLovato** [comment]
I just left comments because it's a solid article overall. If you're fine with the idea, I'd add edits to get closer to the writing guidelines on top of your work, as in #4073

File: tutorials/math/random_number_generation.rst (hunk 2 above)
Comment (Calinou) [reply tutorials/math/random_number_generation.rst:61]: I wrote that section mainly to have an "overview" of all methods you can use. In hindsight, it's true we could move these examples to the class reference, but the class reference isn't always easy to browse when you're looking for various unrelated methods.

**Comment by Calinou** [comment]
Feel free to merge and edit later :slightly_smiling_face:

File: tutorials/math/random_number_generation.rst [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+        else:
+            # 5% chance of being returned.
+            return "Rare"
+
+.. _doc_random_number_generation_shuffle_bags:
+
+"Better" randomness using shuffle bags
+--------------------------------------
+
+Taking the same exemple as above, we would like to pick fruits at random.
```
Comment (NathanLovato) [inline tutorials/math/random_number_generation.rst:228]:
```suggestion
Taking the same example as above, we would like to pick fruits at random.
```

**Comment by NathanLovato** [comment]
Seems I can directly edit the doc and commit to the PR. I'll do so later.

**Comment by NathanLovato** [comment]
The overview of some common functions makes sense after a second read and checking the class reference: it's much more convenient for the reader than having to open two pages and searching for the functions.

**Comment by NathanLovato** [comment]
Thanks for your work!

**Pull request merged**